sql/src/main/java/io/crate/metadata/sys/SysChecksTableInfo.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to Crate.IO GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.metadata.sys;
+
+import com.google.common.collect.ImmutableList;
+import io.crate.analyze.WhereClause;
+import io.crate.core.collections.TreeMapBuilder;
+import io.crate.metadata.*;
+import io.crate.planner.RowGranularity;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+
+import javax.annotation.Nullable;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.List;
+import java.util.Collection;
+import java.util.Iterator;
+
+
+@Singleton
+public class SysChecksTableInfo extends SysTableInfo {
+
+    public static final TableIdent IDENT = new TableIdent(SCHEMA, "checks");
+
+    private static final ImmutableList<ColumnIdent> primaryKey = ImmutableList.of(new ColumnIdent("id"));
+
+    private static final Routing ROUTING = new Routing(
+            TreeMapBuilder.<String, Map<String, List<Integer>>>newMapBuilder().put(
+            NULL_NODE_ID,
+            TreeMapBuilder.<String, List<Integer>>newMapBuilder().put(IDENT.fqn(), null).map()).map()
+    );
+
+    private static final Map<ColumnIdent, ReferenceInfo> INFOS = new LinkedHashMap<>();
```

Comment:
since this is a singleton can we make the attributes here non static? Like how it is done in SysOperationsTableInfo